src/_P057_HT16K33_LED.ino

```diff
@@ -180,23 +180,21 @@ boolean Plugin_057(byte function, struct EventStruct *event, String& string)
       {
         String text = parseStringToEnd(string, 2);
 
-        if (text.length() > 0) {
+        if (!text.isEmpty()) {
           uint8_t seg = 0;
-          bool    setDot = false;
+          uint8_t txt = 0; // Separate indexers for text and segments
+          bool    setDot;
 
           P057_data->ledMatrix.ClearRowBuffer();
 
-          while (text[seg] && seg < 8)
+          while (text[txt] && seg < 8)

```

Comment:
This is tricky, as there is no proper check for `txt < text.length()`

Reply:
It is the same check that was already there, but I've added an extra length check, just to be sure.